src/equation_systems/icns/icns_ops.H

```diff
@@ -80,6 +86,414 @@ struct SrcTermOp<ICNS> : SrcTermOpBase<ICNS>
     }
 };
 
+template<>
+struct AdvectionOp<ICNS, fvm::Godunov>
+{
+    AdvectionOp(PDEFields& fields_in) : fields(fields_in)
+    {
+
+        { // Prefix incflo
+            amrex::ParmParse pp("incflo");
+            pp.query("use_ppm", godunov_ppm);
+            pp.query("godunov_use_forces_in_trans", godunov_use_forces_in_trans);
+        }
+
+        { // Prefix mac
+            amrex::ParmParse pp_mac("mac_proj");
+            pp_mac.query( "mg_rtol", mac_mg_rtol);
+            pp_mac.query( "mg_atol", mac_mg_atol);
+            pp_mac.query( "mg_max_coarsening_level", mac_mg_max_coarsening_level);
+        }
+
+        // TODO: Need iconserv flag to be adjusted???
+        iconserv.resize(ICNS::ndim, 0);
+    }
+
+    void operator()(const FieldState fstate, const amrex::Real dt)
```

Comment:
Eventually we will want to break this into smaller pieces. When we do that we can address the MAC projection duplication between Godunov/MOL.